core/src/main/java/org/mule/DefaultMuleMessage.java

```diff
@@ -485,6 +524,15 @@ public void setProperty(String key, Object value, PropertyScope scope)
         }
     }
 
+    private void updateProperties(String key, Object value)
```

Comment:
This method name is unclear I think.  It's not updating properties, it's updating the dataType, and it's only doing it for a specific property key.